### PR TITLE
Report total filesize in Fido queries

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/sunpy/sunpy/issues/5282. A short example:
```python
from sunpy.net import Fido, attrs as a

result = Fido.search(a.Instrument('AIA'), a.Time('2020-01-01', '2020-01-02'))
print(result)
```

```
Results from 1 Provider:

1328 Results from the VSOClient:
Source: http://vso.stanford.edu/cgi-bin/search
Total size: 3.905 Tbyte
```

### CODE REVIEW

File: sunpy/net/fido_factory.py [hunk 1]
```diff
@@ -183,7 +184,11 @@ def __str__(self):
         for block in self:
             ret += f"{len(block)} Results from the {block.client.__class__.__name__}:\n"
             if block.client.info_url is not None:
-                ret += f'Source: {block.client.info_url}\n\n'
+                ret += f'Source: {block.client.info_url}\n'
+            size = block.total_size()
+            if np.isfinite(size):
+                ret += f'Total size: {size}\n'
```
Comment (Cadair) [inline sunpy/net/fido_factory.py:190]: The VSO isn't always reliable at accurately reporting size, and some clients wont contribute, so I think we should hedge the language a little here?
```suggestion
                ret += f'Total estimated size: {size}\n'
```

File: changelog/5659.feature.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Added the :meth:`~sunpy.net.base_client.QueryResponseTable.total_size`, which
+estimates the total size of the results from a Fido query. If this is supported
+by a client, the total size is printed alongside the results.
+
+To add support for this in external clients, make sure one column contains
```
Comment (Cadair) [inline changelog/5659.feature.rst:5]:
```suggestion
To add support for this in clients, make sure one column contains
```

File: sunpy/net/base_client.py [hunk 3]
```diff
@@ -172,6 +175,8 @@ class QueryResponseTable(QTable):
     display_keys = TableAttribute(default=slice(None))
     hide_keys = TableAttribute()
 
+    size_column = None
+
     # This is a work around for https://github.com/astropy/astropy/pull/11217
     # TODO Remove when min astropy version is > 4.2.1
```
Comment (Cadair) [inline sunpy/net/base_client.py:181]: Oh hey, we can remove this!

**Review by Cadair — APPROVED** [review]

File: changelog/5659.feature.rst [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+the individual filesizes as `~astropy.units.Quantity`, and set the
+``size_column`` class attribute to the name of this column.
```
Comment (nabobalis) [inline changelog/5659.feature.rst:7]: Should there be a note in the dev docs for writing clients about this?

**Review by nabobalis — APPROVED** [review]

File: changelog/5659.feature.rst (hunk 4 above)
Comment (Cadair) [reply changelog/5659.feature.rst:7]: Ah yes, we should add a paragraph to the dev docs about this.

**Review by Cadair — CHANGES_REQUESTED** [review]
plz add dev docs

**Review by Cadair — APPROVED** [review]
Thanks!

**Pull request merged**